WORKSPACE

```diff
@@ -135,3 +143,10 @@ git_repository(
 load("@io_bazel_rules_d//d:d.bzl", "d_repositories")
 
 d_repositories()
+
+# For skylib_library.
+git_repository(
```

Comment:
I think dependents will need these dependencies, so can you add them to the `repositories()` function in `container/container.bzl`?

Reply:
I can do `bazel_skylib`, but apparently trying to do that with `io_bazel` ends up with a chicken-and-egg problem: 

```
ERROR: Failed to load Skylark extension '@io_bazel//tools/build_defs/hash:hash.bzl'.
It usually happens when the repository is not defined prior to being used.
Maybe repository 'io_bazel' was defined later in your WORKSPACE file?
```

Since `WORKSPACE` loads `container.bzl` and that loads `container/image.bzl` and that refers to `io_bazel`, I'm not sure how to extract that nicely into repositories only and structural defines separately. Would you want a `container_repos.bzl` that was specific to `WORKSPACE` includes? This model kinda sucks. I would think it'd want to at least warn you that you were asking for a different (esp. if later) tag or commit instead of the "I just hope somebody includes it" model.

Reply:
Alright, I have some kludges that do this in https://github.com/Capstan/rules_docker/commit/ec938fa1b9bbac998581fa8e3a0b06b7287b4090 It makes there be a separation between Skylark to establish repos and Skylark for rules. This allows the latter to refer to other repositories but ensures that all of those repository names have been defined first. A top-level WORKSPACE must load and invoke the repos macro before trying to load other Skylark in that project.

I'm not sure I like it better, but it works (well, it gets further along in e3e). Also, `/bin/true` doesn't exist by default on an Xcode-laden Mac (it's `/usr/bin/true`), and I'm not sure yet what else doesn't work when locally running e2e test on a Mac.